moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1922,7 +1933,13 @@ double moveit::core::RobotState::computeCartesianPath(const JointModelGroup* gro
 
   // decide how many steps we will need for this trajectory
   double distance = (rotated_target.translation() - start_pose.translation()).norm();
-  unsigned int steps = (test_joint_space_jump ? 5 : 1) + (unsigned int)floor(distance / max_step);
+
+  // If we are testing using the jump threshold, we always want at least MIN_STEPS_FOR_JUMP_THRESH steps
+  unsigned int steps;
+  if (test_joint_space_jump)
+    steps = std::max(MIN_STEPS_FOR_JUMP_THRESH, (std::size_t)floor(distance / max_step) + 1);
```

Comment:
This is redundant now. Could you initialize steps with
`steps = (unsigned int)floor(distance / max_step) + 1;`
and subsequently increase `if (jump_threshold > 0.0 && steps < MIN_STEPS_FOR_JUMP_THRESH)`?